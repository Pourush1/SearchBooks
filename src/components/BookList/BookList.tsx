import React from 'react';
import { IBook } from '../../book-search/BookSearch';
import Book from '../Book/Book';

export interface IBookList {
  allAvailableBooks: IBook[];
  wishList: IBook[];
  addToWishList: (book: IBook) => void;
}

const BookList: React.FC<IBookList> = ({
  allAvailableBooks,
  addToWishList,
  wishList
}) => {
  return (
    <div className="row" data-testid="bookContainer">
      {allAvailableBooks.map((book: IBook, index: number) => {
        const disableAddToWishListButton = wishList.find(
          bookInWishList => bookInWishList.thumbnail === book.thumbnail
        )
          ? true
          : false;
        return (
          <Book
            key={index}
            index={index}
            title={book.title}
            publisher={book.publisher}
            publishedDate={book.publishedDate}
            thumbnail={book.thumbnail}
            addToWishList={addToWishList}
            disableAddToWishListButton={disableAddToWishListButton}
          />
        );
      })}
    </div>
  );
};

export default BookList;
