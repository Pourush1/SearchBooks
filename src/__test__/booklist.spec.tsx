import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';
import BookList from '../components/BookList/BookList';

let documentBody: RenderResult;

describe('<BookList />', () => {
  const props = {
    allAvailableBooks: [
      {
        title: 'Java',
        published: '2016-11-05'
      },
      {
        title: 'Javascipt',
        published: '2017-11-05'
      }
    ],
    wishList: [
      {
        title: 'Java',
        published: '2016-11-05'
      }
    ],
    addToWishList: () => {}
  };

  beforeEach(() => {
    documentBody = render(<BookList {...props} />);
  });

  it('renders correct number of books', () => {
    const bookContainer = screen.getAllByTestId('bookContainer');
    expect(bookContainer).toHaveLength(1);

    // const book = screen.getAllByTestId('book');
    // expect(book).toHaveLength(2);
  });

  it('matches snapshot', () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
