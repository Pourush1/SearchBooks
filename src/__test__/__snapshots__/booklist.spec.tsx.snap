// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookList /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="row"
      data-testid="bookContainer"
    >
      <div
        class="col-sm-3 pb-3"
      >
        <div
          class="card p-3"
        >
          <img
            alt="book cover"
            class="card-img-top mb-3"
          />
          <div
            class="card-body"
          >
            <p
              class="font-weight-bold"
            >
              Java
            </p>
            <p />
            <p />
            <button
              class="btn btn-primary"
              disabled=""
            >
              Add to WishList
            </button>
          </div>
        </div>
      </div>
      <div
        class="col-sm-3 pb-3"
      >
        <div
          class="card p-3"
        >
          <img
            alt="book cover"
            class="card-img-top mb-3"
          />
          <div
            class="card-body"
          >
            <p
              class="font-weight-bold"
            >
              Javascipt
            </p>
            <p />
            <p />
            <button
              class="btn btn-primary"
              disabled=""
            >
              Add to WishList
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
